source/extensions/filters/http/composite/action.cc

```diff
@@ -0,0 +1,22 @@
+#include "extensions/filters/http/composite/action.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Composite {
+absl::Status CompositeAction::createFilters(Http::FilterChainFactoryCallbacks& callbacks) const {
+  // We make use of exceptions here because we don't control the callbacks interface, so we have to
+  // bubble up errors without being able to change the interface.
```

Comment:
Can this happen on the data plane? Just asking as worried about potential new data plane exception opportunity.

Reply:
This would happen on the worker threads during stream processing, but I was thinking that the try catch here would prevent them from escaping this function? Or is there something I'm missing here?

Reply:
We'd like to avoid any exceptions happening on the data plane. There are concerns with exceptions interfering with compiler hardenings such as SLH for NetSpectre, but they become hard to fully rationalize on case-by-case such as here.

Is the issue that we don't want to build the factories until needed, i.e. lazily? It seems the factories could be built but not instantiated on main thread.

Reply:
I already have the factories constructed at config load time, the problem is that by evoking the factories there are a number of factory callbacks that aren't supported: any of the ones that add a match tree, as well as the callback that adds an access logger. I also included a filter factory attempting to create multiple filters as an error condition.

So the problem is that we aren't able to fully validate the configuration at config load due to the `FilterChainFactoryCallbacks` interface allowing too much flexibility at filter creation time.

All that said I can rewrite the current code to set an error object on the factory wrapper instead of using exceptions, which should address the immediate concern of exceptions on the data plane.

Reply:
I think all `FilterChainFactoryCallbacks` should not throw and have a regular return code +1.